lib/daru/dataframe.rb

```diff
@@ -2015,6 +2015,22 @@ def split_by_category cat_name
       end
     end
 
+    # returns array of row tuples at given index(s)
+    def access_row_tuples_by_indexs *indexes
+      positions = @index.pos(*indexes)
+
+      return populate_row_for(positions) if positions.is_a? Numeric
+
+      res = []
+      new_rows = @data.map { |vec| vec[*indexes] }
+      indexes.each do |index|
+        tuples = []
+        new_rows.map { |row| tuples += [row[index]] }
+        res << tuples
+      end
+      res
```

Comment:
```ruby
indexes.map { |index| new_rows.map { |row| row[index] } }
```
(Or something like that, it is hard to guess what data structure you are trying to achieve here, but proper use of `map` will simplify code in any case.)